packages/jest-core/src/TestScheduler.ts

```diff
@@ -193,16 +208,61 @@ export default class TestScheduler {
     if (testsByRunner) {
       try {
         for (const runner of Object.keys(testRunners)) {
-          await testRunners[runner].runTests(
-            testsByRunner[runner],
-            watcher,
-            onStart,
-            onResult,
-            onFailure,
-            {
-              serial: runInBand || Boolean(testRunners[runner].isSerial),
-            },
-          );
+          const testRunner = testRunners[runner];
+          const context = contextsByTestRunner.get(testRunner);
+          const tests = testsByRunner[runner];
+
+          const testRunnerOptions = {
+            serial: runInBand || Boolean(testRunner.isSerial),
+          };
+
+          /**
+           * Test runners with event emitters are still not supported
+           * for third party test runners.
+           */
+          if (
+            testRunner.__PRIVATE_UNSTABLE_API_supportsEventEmmiters__ &&
+            typeof testRunner.UNSTABLE_eventEmitter !== 'undefined'
+          ) {
+            const unsubscribes = [
+              testRunner.UNSTABLE_eventEmitter.on(
+                'test-file-start',
+                ([test]: [TestRunner.Test]) => onTestFileStart(test),
+              ),
+              testRunner.UNSTABLE_eventEmitter.on(
+                'test-file-success',
+                ([test, testResult]: [TestRunner.Test, TestResult]) =>
+                  onResult(test, testResult),
+              ),
+              testRunner.UNSTABLE_eventEmitter.on(
+                'test-file-failure',
+                ([test, error]: [TestRunner.Test, SerializableError]) =>
+                  onFailure(test, error),
+              ),
+              testRunner.UNSTABLE_eventEmitter.on(
+                'test-case-result',
+                ([testPath, testCaseResult]: [Config.Path, AssertionResult]) => {
+                  if (context) {
+                    const test: TestRunner.Test = { context, path: testPath };
+                    this._dispatcher.onTestCaseResult(test, testCaseResult);
+                  }
+                },
+              ),
+            ];
```

Comment:
```suggestion
          if (testRunner.__PRIVATE_UNSTABLE_API_supportsEventEmmiters__) {
            const unsubscribes = [
              testRunner.on('test-file-start', ([test]) =>
                onTestFileStart(test),
              ),
              testRunner.on('test-file-success', ([test, testResult]) =>
                onResult(test, testResult),
              ),
              testRunner.on('test-file-failure', ([test, error]) =>
                onFailure(test, error),
              ),
              testRunner.on(
                'test-case-result',
                ([testPath, testCaseResult]) => {
                  if (context) {
                    const test: TestRunner.Test = {context, path: testPath};
                    this._dispatcher.onTestCaseResult(test, testCaseResult);
                  }
                },
              ),
            ];
```

the type stuff in the emitter works! So we don't need to repeat the type in the listener function anymore 🎉